packages/common-common/src/permissions.ts

```diff
@@ -1,63 +1,123 @@
+  [Action.EDIT_THREAD, [Action.CREATE_REACTION, Action.VIEW_REACTIONS, Action.CREATE_COMMENT, Action.VIEW_COMMENTS, Action.EDIT_COMMENT, Action.DELETE_COMMENT, Action.CREATE_POLL, Action.VIEW_POLLS, Action.VOTE_ON_POLLS, Action.VIEW_THREADS, Action.CREATE_THREAD, Action.DELETE_THREAD]],
+  [Action.DELETE_THREAD, [Action.CREATE_REACTION, Action.VIEW_REACTIONS, Action.CREATE_COMMENT, Action.VIEW_COMMENTS, Action.EDIT_COMMENT, Action.DELETE_COMMENT, Action.CREATE_POLL, Action.VIEW_POLLS, Action.VOTE_ON_POLLS, Action.VIEW_THREADS, Action.CREATE_THREAD, Action.EDIT_THREAD]],
+]);
+
 export type Permissions = bigint;
 
 export enum PermissionError {
   NOT_PERMITTED = 'Action not permitted',
 }
 
+// Must be Used to Add Permissions
 export function addPermission(
   permission: Permissions,
   actionNumber: number
 ): Permissions {
   let result = BigInt(permission);
   // eslint-disable-next-line no-bitwise
   result |= BigInt(1) << BigInt(actionNumber);
+  const implicitActions = IMPLICIT_PERMISSIONS_BY_ACTION.get(actionNumber);
+  if (implicitActions) {
+    for (let i = 0; i < implicitActions.length; i++) {
+      // eslint-disable-next-line no-bitwise
+      result |= BigInt(1) << BigInt(Number(implicitActions[i]));
+    }
+    return result;
+  }
   return result;
 }
 
+// Must be Used to Remove Permissions
 export function removePermission(
   permission: Permissions,
   actionNumber: number
 ): Permissions {
   let result = BigInt(permission);
   // eslint-disable-next-line no-bitwise
   result &= ~(BigInt(1) << BigInt(actionNumber));
+  const implicitActions = IMPLICIT_PERMISSIONS_BY_ACTION.get(actionNumber);
+  if (implicitActions) {
+    for (let i = 0; i < implicitActions.length; i++) {
+      result &= ~(BigInt(1) << BigInt(Number(implicitActions[i])));
+    }
+  }
   return result;
 }
 
+// Default Permissions
 export const BASE_PERMISSIONS: Permissions =
   addPermission(BigInt(0), Action.CREATE_THREAD) |
-  addPermission(BigInt(0), Action.VIEW_CHAT_CHANNELS);
+  addPermission(BigInt(0), Action.VIEW_CHAT_CHANNELS) |
+  addPermission(BigInt(0), Action.VIEW_THREADS);
 
+// Checks if a permission has a specific action
 export function isPermitted(permission: Permissions, action: number): boolean {
   const actionAsBigInt: bigint = BigInt(1) << BigInt(action);
   const hasAction: boolean =
     (BigInt(permission) & actionAsBigInt) == actionAsBigInt;
   return hasAction;
 }
 
+// Adds the implicit permissions to a permission
+export function computeImplicitPermissions(
+  permission: Permissions
+): Permissions {
+  let result = BigInt(permission);
+  // Find the highest hierachy action that is permitted
+  for (const [action, implicitActions] of IMPLICIT_PERMISSIONS_BY_ACTION) {
```

Comment:
Why do we need to loop over the map? Shouldn't we be able to know which entries on the map are meaningful based on the provided permission?